lib/r10k/git/stateful_repository.rb

```diff
@@ -31,6 +31,10 @@ def initialize(remote, basedir, dirname)
   end
 
   def resolve(ref)
+    if ref == :latest
+      @cache.sync!
+      ref = "HEAD"
+    end
     @cache.sync if sync_cache?(ref)
```

Comment:
Shouldn't this line already be doing what the stanza above does?

1. This calls `#sync` on the cache IF `#sync_cache?` returns true for `ref`.  
Note that `R10K::Git::Cache#sync` is just a wrapper for `R10K::Git::Cache#sync!` with a guard against multiple sync activities in the same r10k run.
2. `#sync_cache?` [returns true](https://github.com/puppetlabs/r10k/blob/e27a5d14c05324a89820ef8c382f63cb407410fb/lib/r10k/git/stateful_repository.rb#L96) if the cache doesn't exist, or if the ref type is NOT `:commit` or `:tag`, according to  [`R10K::Git::Cache#ref_type`](https://github.com/puppetlabs/r10k/blob/7d8b5b51b533ea3c45016d54e837aecc6f657588/lib/r10k/git/rugged/base_repository.rb#L36-L49).
3. Therefore, `@cache.sync` should already be being called here. If it isn't, that's a bug that needs to be hunted down and fixed, not a change in intended behavior.